sphinx/user-guide/installation.rst

```diff
@@ -0,0 +1,427 @@
+most installations are likely to be more complex. There are five types of
+installation:
+
+Hosts For Running Cylc Suites
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Each user can just run their suites on the host where they issue the
+``rose suite-run`` command. However, the local host may not meet the
+necessary requirements for connectivity (to the hosts running the tasks)
+or for availability (the host needs to remain up for the duration of
+the suite). Therefore, Rose can be configured to run the suites on
+remote hosts. If multiple hosts are available, by default the least loaded
+of the available hosts will be used.
+
+Installation requirements:
+   * Rose, cylc, Bash, Python, jinja2.
+   * Subversion & FCM *(only if you want rose suite-run to install files
+     from Subversion using FCM keywords).*
+Connectivity requirements:
+   * Must be able to submit tasks to the hosts which run the suite tasks,
+     either directly or via SSH access to another host.
+
+Hosts For Running Tasks In Suites
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Installation requirements:
+   * Rose, cylc, Bash, Python.
+   * Subversion & FCM (only if you want to use the Rose install utility
+     to install files from Subversion using FCM keywords).
+
+Connectivity requirements:
+   * Must be able to communicate back to the hosts running the cylc suites
+     via HTTPS (preferred), HTTP or SSH.
+
+Hosts For Running User Interactive Tools
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Installation requirements:
+   * Rose, cylc, Bash, Python, requests, Subversion, FCM,
+     Pygraphviz (+ graphviz), PyGTK (+ GTK).
+
+Connectivity requirements:
+   * Must have HTTP access to the hosts running the Rosie web service.
+   * Must have access to the Rosie Subversion repositories via the
+     appropriate protocol.
+   * Must have HTTP and SSH access to the hosts running the cylc suites.
+   * Must share user accounts and $HOME directories with the hosts running
+     the cylc suites.
+
+Hosts For Running Rose Bush
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Installation requirements:
+   * Rose, Bash, Python, cherrypy, jinja2.
+
+Connectivity requirements:
+   * Must be able to access the home directories of users' cylc run directories.
+
+Hosts For Rosie Subversion Repositories And The Rosie web Services
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Typically you will only need a single host but you can have multiple
+repositories on different hosts if you require this.
+
+Installation requirements:
+   * Rose, Bash, Python, cherrypy, jinja2, sqlalchemy, Subversion.
+
+.. note::
+
+   This section has assumed that you wish to use Rose & cylc (including
+   their respective GUIs) and use Rosie to store your suites. However,
+   there are many ways of using Rose. For instance:
+
+   * You can use Rose without using cylc.
+   * You can use Rose & cylc but not use Rosie.
+   * You can use Rose & cylc from the command line without using their GUIs.
+
+
+Configuring Rose
```

Comment:
Would you like me to do a functional review for the installation commands/instructions? Currently just eyeballing them to check they are sensible.

Reply:
It would be difficult to perform a functional review as it is site configuration. We should perform a larger review of the instructions on this page at some point.

Reply:
True.